src/lib/layers/editable-geojson-layer.js

```diff
@@ -215,78 +224,112 @@ export default class EditableGeoJsonLayer extends CompositeLayer {
   }
 
   onPointerMove(event) {
-    if (this.state.draggingPoint) {
-      // stop propagation to prevent map panning
-      event.stopPropagation();
-
-      if (this.props.onDraggingPoint) {
-        const pointerCoords = this.getPointerCoords(event);
-        const groundCoords = this.context.viewport.unproject([pointerCoords.x, pointerCoords.y]);
-
-        const { coordinateIndexes } = this.state.draggingPoint;
-        const editingFeature = this.getEditingFeature();
-        const isPolygonal =
-          editingFeature.geometry.type === 'Polygon' ||
-          editingFeature.geometry.type === 'MultiPolygon';
-
-        const coordinates = immutablyReplaceCoordinate(
-          editingFeature.geometry.coordinates,
-          coordinateIndexes,
-          groundCoords,
-          isPolygonal
-        );
-
-        const updatedFeature = {
-          ...editingFeature,
-          geometry: {
-            ...editingFeature.geometry,
-            coordinates
-          }
-        };
-
-        this.props.onDraggingPoint({
-          feature: updatedFeature,
+    if (!this.state.pointerDownPosition) {
+      // TODO: only subscribe to pointermove once the pointer goes down (at which point we can remove this check)
+      return;
+    }
+
+    // stop propagation to prevent map panning
+    event.stopPropagation();
+
+    const pointerCoords = this.getPointerCoords(event);
+
+    if (!this.state.draggingPosition) {
+      // pointer is moving, but is it moving enough?
+
+      if (
+        Math.abs(this.state.pointerDownCoords.x - pointerCoords.x) >
+          MINIMUM_POINTER_MOVE_THRESHOLD_PIXELS ||
+        Math.abs(this.state.pointerDownCoords.y - pointerCoords.y) >
+          MINIMUM_POINTER_MOVE_THRESHOLD_PIXELS
+      ) {
+        this.state.draggingPosition = this.state.pointerDownPosition;
+
+        // Fire the start dragging event
+        (this.props.onStartDraggingPosition || (() => {}))({
           featureIndex: this.props.selectedFeatureIndex,
-          coordinateIndexes: this.state.draggingPoint.coordinateIndexes
+          positionIndexes: this.state.draggingPosition.positionIndexes
         });
+      } else {
+        // pointer barely moved, so nothing else to do
+        return;
       }
     }
+
+    const position = this.context.viewport.unproject([pointerCoords.x, pointerCoords.y]);
+    const { positionIndexes } = this.state.draggingPosition;
+    const { selectedFeatureIndex } = this.props;
+
+    const updatedData = this.state.editableFeatureCollection
+      .replacePosition(selectedFeatureIndex, positionIndexes, position)
+      .getObject();
+
+    (this.props.onEdit || (() => {}))({
```

Comment:
Since it is stateless and an immutable update, the caller chooses to reject the edit by not updating their own state. I'll add a demo of what I mean to the example app.

Reply:
Ohh ok.